build: fetch macOS Homebrew bottles from GitHub Packages instead of bintray

michaelsbradleyjr opened the pull request:
bintray (for open source) is shutting down permanently on May 1:

https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/

At this time downloading from GitHub Packages requires specifying a username and personal acccess token, but it seems they don't have to be valid. For example, in the changes to the Makefile I specified `_:_` as the username and token.

Also, change the minimum macOS version from 10.13 to 10.14 since Homebrew bottles are no longer built for 10.13 (High Sierra):

https://github.com/Homebrew/discussions/discussions/1264#discussioncomment-601544

Closes #1851

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 4a2f408b | [#1](https://ci.status.im/job/status-desktop/job/macos/job/PR-2244/1/) | 2021-04-13 19:16:19 | ~5 min | `macos` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/macos/job/PR-2244/1/consoleText) |

michaelsbradleyjr commented:
@jakubgs the build is failing in Jenkins on macOS (this PR is only relevant for macOS).  It may be that the `jq` cli tool needs to be installed in the environment (e.g. `brew install jq`).

Or it may be that my trick of specifying `-u _:_` for `curl` works locally but not in CI and we'll need to setup a GitHub username and access token in environment variables.

michaelsbradleyjr commented:
@jakubgs I've temporarily added `V=1` to `make deps` and `-v` to `curl`:

https://ci.status.im/job/status-desktop/job/macos/job/PR-2244/4/consoleText

So `jq` is there and working properly, but note the URLs resolved from the output of `brew info` are pointing to `homebrew.bintray.com`.

That means that on the macOS host, `brew` needs to be updated to the latest version (3.1.1), which includes the changeover from bintray hosting to GitHub Packages hosting.

jakubgs commented:
Done and done:
```
 > ansible ci-slave-macos -o -a '/usr/local/bin/brew --version | head -n1' 
macos-02.ms-eu-dublin.ci.misc | CHANGED | rc=0 | (stdout) Homebrew 3.1.1
macos-01.ms-eu-dublin.ci.misc | CHANGED | rc=0 | (stdout) Homebrew 3.1.1
macos-03.ms-eu-dublin.ci.misc | CHANGED | rc=0 | (stdout) Homebrew 3.1.1
```

jakubgs commented:
@michaelsbradleyjr My dude, you can just modify the last commit without actually modifying it, hence changing the SHA:
```
git commit --amend --no-edit
```
I have this in my `~/.gitconfig`:
```gitconfig
[alias]
    c   = commit --verbose --no-verify --gpg-sign --signoff
    m   = commit --verbose --no-verify --gpg-sign --signoff --amend
    mm  = commit --verbose --no-verify --gpg-sign --signoff --amend --no-edit
    fix = commit --verbose --no-verify --gpg-sign --signoff --fixup
```

michaelsbradleyjr commented:
> @michaelsbradleyjr My dude, you can just modify the last commit without actually modifying it, hence changing the SHA:

Thanks for the tip, but I'm not sure I completely understand. I dropped the three WIP commits and then rebased on `master` because I wanted to run the CI build locally one more time, i.e. after it includes changes on `master` that landed since I opened the PR.

Before I rebased, i.e. had only dropped the WIP commits, the SHA for `build: fetch macOS Homebrew bottles from GitHub Packages instead of bintray` indeed didn't change.

jakubgs commented:
I'm referring to the bump commit: https://github.com/status-im/status-desktop/pull/2244/commits/eb66cc1fcba789378c4eb8871b96fff736954b6f
Which seems to be gone now, but when I looked it just contained an empty file:
![image](https://user-images.githubusercontent.com/2212681/114756100-ad987780-9d5a-11eb-9c2c-1cdc21fe1f50.png)

michaelsbradleyjr commented:
> I'm referring to the bump commit: [eb66cc1](https://github.com/status-im/status-desktop/commit/eb66cc1fcba789378c4eb8871b96fff736954b6f)
> Which seems to be gone now, but when I looked it just contained an empty file:
> ![image](https://user-images.githubusercontent.com/2212681/114756100-ad987780-9d5a-11eb-9c2c-1cdc21fe1f50.png)

Ah, sorry, I misunderstood what you were referring to. Yes, what you suggested is a better way to do it (bumping to trigger build), I'll adapt my workflow.
